packages/axelar-std/src/events.rs

```diff
@@ -56,107 +56,64 @@ mod testutils {
-                #[allow(unused_mut)]
-                fn standardized_fmt(env: &soroban_sdk::Env, (contract_id, topics, data): &(soroban_sdk::Address, soroban_sdk::Vec<soroban_sdk::Val>, soroban_sdk::Val)) -> std::string::String {
-                    use soroban_sdk::TryFromVal;
-
-                    let mut topics_output: std::vec::Vec<std::string::String> = std::vec![];
-
-                    let mut i = 0;
-                    $(
-                        let topic = topics.get(i).expect("the number of topics does not match this function's definition");
-                        topics_output.push(std::format!("{:?}", <$topic_type>::try_from_val(env, &topic)
-                            .expect("given topic value does not match the expected type")));
-
-                        i += 1;
-                    )*
-
-                    let data = soroban_sdk::Vec::<soroban_sdk::Val>::try_from_val(env, data).expect("data should be defined as a vector-compatible type");
-
-                    let mut data_output: std::vec::Vec<std::string::String> = std::vec![];
-
-                    let mut i = 0;
-                    $(
-                        let data_entry = data.get(i).expect("the number of data entries does not match this function's definition");
-                        data_output.push(std::format!("{:?}", <$data_type>::try_from_val(env, &data_entry)
-                            .expect("given data value does not match the expected type")));
-
-                        i += 1;
-                    )*
-
-                    std::format!("contract: {:?}\ntopics: ({})\ndata: ({})", contract_id, topics_output.join(", "), data_output.join(", "))
-                }
-            }
-        };
-    }
 }
 
 #[cfg(test)]
 mod test {
     use core::fmt::Debug;
 
-    use soroban_sdk::testutils::Events;
     use soroban_sdk::xdr::Int32;
-    use soroban_sdk::{contract, BytesN, Env, IntoVal, String, Symbol, Topics, Val};
+    use soroban_sdk::{contract, contractimpl, BytesN, Env, String, Symbol};
+    use stellar_axelar_std_derive::IntoEvent;
+    use stellar_axelar_std::events::Event;
 
-    use crate::events::{Event, EventTestutils};
-    use crate::{events, impl_event_testutils};
+    use crate::events::fmt_last_emitted_event;
 
-    #[derive(Debug, PartialEq, Eq, Clone)]
+    #[derive(Debug, PartialEq, Eq, IntoEvent)]
     struct TestEvent {
         topic1: Symbol,
         topic2: String,
         topic3: Int32,
+        #[data]
         data1: String,
+        #[data]
         data2: BytesN<32>,
     }
 
-    impl Event for TestEvent {
-        fn topics(&self, _env: &Env) -> impl Topics + Debug {
-            (self.topic1.clone(), self.topic2.clone(), self.topic3)
-        }
+    #[contract]
+    struct Contract;
 
-        fn data(&self, _env: &Env) -> impl IntoVal<Env, Val> + Debug {
-            (self.data1.clone(), self.data2.clone())
+    #[contractimpl]
+    impl Contract {
+        pub fn emit_test_event(env: &Env) {
```

Comment:
we already added integration tests for this, so a unit test isn't needed. this should resolve the dependency issue as well i believe